### PR TITLE
#2615 corrections unit tests

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#2615

*Description of changes:*

* Fixes Existing unit testing for components affected by Corrections

*Implements or updates unit tests for the following:*

* DetailsList ( new component )
* Filing History List
* ToDo List

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by cameron-freshworks** [comment]
> Looks great! Isn't testing small components fun?
> 
> Just a couple of comments that would be nice to implement.

haha love it :)

File: tests/unit/DetailsList.spec.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+          'filingId': 63958,
+          'id': 123,
+          'submitterDisplayName': 'cbIdIr1234',
+          'timestamp': '2020-03-02T20:26:31.697044+00:00'
+        }
+      }
+    ]
+  }
+
+  const mockManyCommentsFiling = {
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply tests/unit/DetailsList.spec.ts]: Quite literal isn't it 😆

File: tests/unit/DetailsList.spec.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+          'comment': 'Correction for Annual Report (2018). Filed on 2018-01-08.',
+          'filingId': 63958,
+          'id': 111,
+          'submitterDisplayName': 'cbIdIr1234',
+          'timestamp': '2020-03-02T20:26:31.697044+00:00'
+        }
+      },
+      {
+        'comment': {
+          'comment': 'Correction for Director Change. Filed on 2018-01-08.',
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply tests/unit/DetailsList.spec.ts]: Makes perfect sense, will update this!

File: tests/unit/DetailsList.spec.ts [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+          'filingId': 63958,
+          'id': 333,
+          'submitterDisplayName': 'cbIdIr1234',
+          'timestamp': '2020-03-06T20:26:33.697044+00:00'
+        }
+      }
+    ]
+  }
+
+  beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply tests/unit/DetailsList.spec.ts]: 100%, i will correct this 👍

File: tests/unit/DetailsList.spec.ts [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    expect(vm.$el.querySelector('.comments-section')).toBeNull()
+
+    wrapper.destroy()
+  })
+
+  it('Displays the comments section when comments are present in the filing', () => {
+    const wrapper = shallowMount(DetailsList, { store, propsData: { filing: mockFiling } })
+    const vm: any = wrapper.vm
+
+    expect(vm.$el.querySelector('.comments-section')).not.toBeNull()
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply tests/unit/DetailsList.spec.ts]: Good idea, i will add this!

File: tests/unit/DetailsList.spec.ts [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+Vue.use(Vuetify)
+Vue.use(Vuelidate)
+const vuetify = new Vuetify({})
+
+describe('Details List', () => {
+  const mockNoCommentsFiling = {
+    'type': 'annualReport',
+    'title': 'Annual Report (2018)',
+    'filingId': 63958,
+    'filingAuthor': 'Registry Staff',
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply tests/unit/DetailsList.spec.ts]: Oops, yes that could be misleading in some scenarios, will correct this.

File: tests/unit/DetailsList.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply tests/unit/DetailsList.spec.ts]: Added a new test for this! 👍

**Comment by cameron-freshworks** [comment]
Updates made!

File: tests/unit/DetailsList.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply tests/unit/DetailsList.spec.ts]: Nice!

**Pull request merged**